src/main/java/mate/academy/bookstore/service/impl/OrderServiceImpl.java

```diff
@@ -0,0 +1,99 @@
+import mate.academy.bookstore.model.Order;
+import mate.academy.bookstore.model.OrderItem;
+import mate.academy.bookstore.model.ShoppingCart;
+import mate.academy.bookstore.model.User;
+import mate.academy.bookstore.repository.cartitem.CartItemRepository;
+import mate.academy.bookstore.repository.order.OrderRepository;
+import mate.academy.bookstore.repository.orderitem.OrderItemRepository;
+import mate.academy.bookstore.repository.shoppingcart.ShoppingCartRepository;
+import mate.academy.bookstore.repository.user.UserRepository;
+import mate.academy.bookstore.service.OrderService;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+@Service
+@RequiredArgsConstructor
+public class OrderServiceImpl implements OrderService {
+    private static final String CANNOT_FIND_USER_BY_ID = "Can't find user by id: ";
+    private static final String CANNOT_FIND_ORDER_USER_ID = "Can't find order by user id: ";
+    private static final String CANNOT_FIND_SHOPPING_CART_BY_USER_ID
+            = "Can't find shopping cart by user id: ";
+    private final OrderRepository orderRepository;
+    private final ShoppingCartRepository shoppingCartRepository;
+    private final UserRepository userRepository;
+    private final OrderItemRepository orderItemRepository;
+    private final CartItemRepository cartItemRepository;
+    private final OrderMapper orderMapper;
+
+    @Override
+    @Transactional
+    public OrderResponseDto placeOrder(AddressRequestDto requestDto, Long userId) {
+        ShoppingCart shoppingCart = shoppingCartRepository.getShoppingCartByUserId(userId)
+                .orElseThrow(() -> new EntityNotFoundException(
+                        CANNOT_FIND_SHOPPING_CART_BY_USER_ID + userId));
+
+        User user = userRepository.findById(userId)
+                .orElseThrow(() -> new EntityNotFoundException(CANNOT_FIND_USER_BY_ID + userId));
+
+        Order order = new Order();
+
+        BigDecimal total = shoppingCart.getCartItems().stream()
+                .map(item -> item.getBook().getPrice()
+                        .multiply(BigDecimal.valueOf(item.getQuantity())))
+                .reduce(BigDecimal.ZERO, BigDecimal::add);
+
+        order.setTotal(total);
+        order.setUser(user);
+        order.setShippingAddress(requestDto.shippingAddress());
+        order.setStatus(Order.Status.PENDING);
+        order = orderRepository.save(order);
+
+        Set<OrderItem> orderItems = new HashSet<>();
+        for (CartItem cartItem : shoppingCart.getCartItems()) {
+            OrderItem orderItem = new OrderItem();
+
+            orderItem.setOrder(order);
+            orderItem.setBook(cartItem.getBook());
+            orderItem.setPrice(cartItem.getBook().getPrice());
+            orderItem.setQuantity(cartItem.getQuantity());
+            orderItemRepository.save(orderItem);
+            orderItems.add(orderItem);
+        }
+
+        order.setOrderItems(orderItems);
```

Comment:
Need to check, but maybe it will have more performance
```suggestion

            orderItems.add(orderItem);
        }
            orderItemRepository.saveAll(orderItems);
        order.setOrderItems(orderItems);
```